File: Sms.Core/Domain/Entities/Uploads/UploadType.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sms.Core.Domain.Entities.Uploads
{
    [Table("upload_types")]
    public class UploadType:BaseEntity<Guid>
    {
        public string Name { get; set; }
        public string Code { get; set; }
        public string Description { get; set; }
        public string FileExtensions { get; set; }
    }
}

File: sms.client/src/app/shared/models/uploads/upload-type.ts
export interface UploadType {
    
    id: string;
    name: string;
    code: string;
    description: string;
    allowedFileTypes: string;
    extensions: UploadMimeType[];
}

export interface UploadMimeType {
    id: number;
	type: string;
	name: string;
	checked: boolean;
}